samples-lib/build.gradle

```diff
@@ -30,4 +34,12 @@ dependencies {
     implementation "androidx.constraintlayout:constraintlayout:$constraintLayoutVersion"
     implementation "com.google.android.material:material:$materialVersion"
     implementation "commons-io:commons-io:2.11.0"
+
+    // lib dependencies for samples using Jetpack Compose
+    implementation "androidx.core:core-ktx:$ktxAndroidCore"
+    implementation "androidx.activity:activity-compose:$composeActivityVersion"
+    implementation "androidx.compose.ui:ui:$composeUI"
+    implementation "androidx.compose.material:material:$composeMaterial"
```

Comment:
Consider using material3. There are some differences in how things work between material and material3, and material is already out of date. It is more likely as time goes on that a user will be porting a sample to material3, and using it here can cut out a lot of friction for them.

Reply:
consider using [the compose BOM](https://developer.android.com/jetpack/compose/bom) to specify a version for all compose dependencies.

Reply:
We plan to tackle this in the a future issue planned for samples. There are some conflicts with just switching the gradle to use Material3 and would be best to resolve them separately before updating the script.

Reply:
Not necessarily, the theming issue is the next planned issue for the compose effort for samples and shouldn't have much affect on the script used in this PR. 

Updating to use Material3 would require changes to our `sampleslib/theme/Theme.kt` file, which would be helpful to resolve as part of the theme issue. The gradle implementation for Material3 is specific for only composable samples: `androidx.compose.material3:material3:1.1.0`. And the gradle implementation for traditional Android UI samples currently is: `com.google.android.material:material:1.7.0`. It seems like the distinction in these two material libraries shouldn't cause changes to update all the samples, just the samples using compose. 

I'm getting dependency resolution errors with with just adding material3 to the project, which tells me that there needs to be an update with the versioning used for our libraries:
```console
Duplicate class kotlin.collections.jdk8.CollectionsJDK8Kt found in modules jetified-kotlin-stdlib-1.8.10 (org.jetbrains.kotlin:kotlin-stdlib:1.8.10) and jetified-kotlin-stdlib-jdk8-1.7.20 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.20)
```